### PR TITLE
Update Scala & ZIO versions 

### PR DESCRIPTION
**Pull request by guizmaii**

### CODE REVIEW

**Review by adamgfraser — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ env:
 on:
   pull_request:
   push:
-    branches: ['master', 'zio-2']
+    branches: ['master', 'series/2.x']
```
Comment (adamgfraser) [inline .github/workflows/ci.yml]: I think can just do `series/2x` at this point.

Comment (sideeffffect) [reply .github/workflows/ci.yml]: Theoretically we could want to publish something in the `series/1.x` branch, right?

Comment (sideeffffect) [inline .github/workflows/ci.yml]:
```suggestion
    branches: ['series/1.x', 'series/2.x']
```

Comment (adamgfraser) [reply .github/workflows/ci.yml]: I don't think we should be publishing snapshots for it.

Comment (sideeffffect) [reply .github/workflows/ci.yml]: I don't mind if we don't do that :shrug: :+1:

File: core/shared/src/main/scala/zio/prelude/AssociativeFlatten.scala [hunk 2]
```diff
@@ -72,17 +72,17 @@ object AssociativeFlatten {
   /**
    * The `AssociativeFlatten` instance for `Const`.
    */
-  implicit def ConstAssociativeFlatten[A]: AssociativeFlatten[({ type ConstA[+B] = Const[A, B] })#ConstA] =
-    new AssociativeFlatten[({ type ConstA[+B] = Const[A, B] })#ConstA] {
+  implicit def ConstAssociativeFlatten[A]: AssociativeFlatten[Const[A, +*]] =
+    new AssociativeFlatten[Const[A, +*]] {
```
Comment (sideeffffect) [inline core/shared/src/main/scala/zio/prelude/AssociativeFlatten.scala]: We may want to choose to switch to kind-projector syntax, but I'd prefer we do it in a dedicated PR (the manual syntax is used _pervasively_).
There also may be some issues related to this new syntax -- there were reasons to not use it in the beginning after all...

**Comment by adamgfraser** [comment]
@guizmaii Can we just keep this PR focused on updating dependencies?

**Comment by guizmaii** [comment]
> Can we just keep this PR focused on updating dependencies?

Yup, sorry about that. Rollbacked

**Review by adamgfraser — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ env:
 on:
   pull_request:
   push:
-    branches: ['master', 'zio-2']
+    branches: ['series/1.x', 'series/2.x']
```
Comment (adamgfraser) [inline .github/workflows/ci.yml]: Weren't we going to change this to just `series/2.x`?

**Review by adamgfraser — CHANGES_REQUESTED** [review]
Just one change on the snapshot publishing.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (guizmaii) [reply .github/workflows/ci.yml]: @adamgfraser I just applied this @sideeffffect proposal: https://github.com/zio/zio-prelude/pull/1048#pullrequestreview-1242270136

Comment (adamgfraser) [reply .github/workflows/ci.yml]: @guizmaii I think based on that conversation we said we weren't going to include `series/1.x`.

Comment (guizmaii) [reply .github/workflows/ci.yml]: @adamgfraser Done :)

**Review by sideeffffect — APPROVED** [review]
Thanks for all the love :heart:

**Review by adamgfraser — APPROVED** [review]
